fs/move.py

```diff
@@ -62,17 +66,42 @@ def move_file(
                 _src_fs.move(
                     src_path, dst_path, overwrite=True, preserve_time=preserve_time
                 )
-            else:
-                # Standard copy and delete
-                with _src_fs.lock(), _dst_fs.lock():
-                    copy_file(
-                        _src_fs,
-                        src_path,
-                        _dst_fs,
-                        dst_path,
-                        preserve_time=preserve_time,
-                    )
-                    _src_fs.remove(src_path)
+                return
+
+            if _src_fs.hassyspath(src_path) and _dst_fs.hassyspath(dst_path):
```

Comment:
Would
```python
if thing1:
    do1
    return

if thing2:
    do2
```
be better as:
```python
if thing1:
    do1
elif thing2:
    do2
```

?

Reply:
This works for the first case, but the second one can pass or return and we need to fall through to the standard method. It comes down to whether you like early exits 🤷‍♂️

Reply:
I'll leave that up to @althonos as he's much more familiar with the general codebase here than I am.

Reply:
I'm not super opinionated about early exits in general, I guess I'm fine with the early return because the second branch is actually also an early return on success. I'd have thought an `if/elif/else` to be better if every branch was expected to succeed, which is not the case here.